make/compiler_flags

```diff
@@ -232,50 +232,55 @@ endif
 # should contain "tbb" or "tbb tbbmalloc tbbmalloc_proxy". Setting the
 # TBB_LIBRARIES variable overrides the default.
 
-TBB_BIN ?= $(MATH)lib/tbb
-TBB_RELATIVE_PATH ?= ../$(notdir $(TBB))
-TBB_BIN_ABSOLUTE_PATH = $(abspath $(TBB_BIN))
-TBB_ABSOLUTE_PATH = $(abspath $(TBB))
-
-ifeq ($(OS),Darwin)
-  TBB_LIBRARIES ?= tbb tbbmalloc tbbmalloc_proxy
-else
-  TBB_LIBRARIES ?= tbb
-endif
+ifndef SYS_LIBS
+  TBB_BIN ?= $(MATH)lib/tbb
+  TBB_RELATIVE_PATH ?= ../$(notdir $(TBB))
+  TBB_BIN_ABSOLUTE_PATH = $(abspath $(TBB_BIN))
+  TBB_ABSOLUTE_PATH = $(abspath $(TBB))
+
+  ifeq ($(OS),Darwin)
+    TBB_LIBRARIES ?= tbb tbbmalloc tbbmalloc_proxy
+  else
+    TBB_LIBRARIES ?= tbb
+  endif
 
-ifdef TBB_LIB
+  ifdef TBB_LIB
 
-ifdef TBB_INC
-CXXFLAGS_TBB ?= -I $(TBB_INC)
-else
-CXXFLAGS_TBB ?= -I $(TBB)/include
-endif
+  ifdef TBB_INC
+  CXXFLAGS_TBB ?= -I $(TBB_INC)
+  else
+  CXXFLAGS_TBB ?= -I $(TBB)/include
+  endif
 
-LDFLAGS_TBB ?= -Wl,-L,"$(TBB_LIB)" -Wl,-rpath,"$(TBB_LIB)" -ltbb
-LDLIBS_TBB ?= $(LDFLAGS_TBB)
+  LDFLAGS_TBB ?= -Wl,-L,"$(TBB_LIB)" -Wl,-rpath,"$(TBB_LIB)" -ltbb
+  LDLIBS_TBB ?= $(LDFLAGS_TBB)
 
-else
+  else
 
-ifeq ($(OS),Windows_NT)
-  TBB_TARGETS ?= $(addprefix $(TBB_BIN)/,$(addsuffix $(LIBRARY_SUFFIX),$(TBB_LIBRARIES)))
-endif
-ifeq ($(OS),Darwin)
-  TBB_TARGETS ?= $(addprefix $(TBB_BIN)/lib,$(addsuffix $(LIBRARY_SUFFIX), $(TBB_LIBRARIES)))
-endif
-ifeq ($(OS),Linux)
-  # Update the suffix with the internal TBB source code!
-  # The new version of TBB library is 12+ (e.g., libtbb.so.12.1 not libtbb.so.2)
-  TBB_TARGETS ?= $(addprefix $(TBB_BIN)/lib,$(addsuffix $(LIBRARY_SUFFIX).2,$(TBB_LIBRARIES)))
-endif
+  ifeq ($(OS),Windows_NT)
+    TBB_TARGETS ?= $(addprefix $(TBB_BIN)/,$(addsuffix $(LIBRARY_SUFFIX),$(TBB_LIBRARIES)))
+  endif
+  ifeq ($(OS),Darwin)
+    TBB_TARGETS ?= $(addprefix $(TBB_BIN)/lib,$(addsuffix $(LIBRARY_SUFFIX), $(TBB_LIBRARIES)))
+  endif
+  ifeq ($(OS),Linux)
+    # Update the suffix with the internal TBB source code!
+    # The new version of TBB library is 12+ (e.g., libtbb.so.12.1 not libtbb.so.2)
+    TBB_TARGETS ?= $(addprefix $(TBB_BIN)/lib,$(addsuffix $(LIBRARY_SUFFIX).2,$(TBB_LIBRARIES)))
+  endif
 
-CXXFLAGS_TBB ?= -I $(TBB)/include
-LDFLAGS_TBB ?= -Wl,-L,"$(TBB_BIN_ABSOLUTE_PATH)" -Wl,-rpath,"$(TBB_BIN_ABSOLUTE_PATH)" $(LDFLAGS_FLTO_FLTO) $(LDFLAGS_OPTIM_TBB)
-LDLIBS_TBB ?= $(LDFLAGS_TBB)
+  CXXFLAGS_TBB ?= -I $(TBB)/include
+  LDFLAGS_TBB ?= -Wl,-L,"$(TBB_BIN_ABSOLUTE_PATH)" -Wl,-rpath,"$(TBB_BIN_ABSOLUTE_PATH)" $(LDFLAGS_FLTO_FLTO) $(LDFLAGS_OPTIM_TBB)
+  LDLIBS_TBB ?= $(LDFLAGS_TBB)
 
-endif
+  endif
 
-ifdef TBB_INTERFACE_NEW
-CPPFLAGS_TBB ?= -DTBB_INTERFACE_NEW
+  ifdef TBB_INTERFACE_NEW
+  CPPFLAGS_TBB ?= -DTBB_INTERFACE_NEW
+  endif
+else
+  CXXFLAGS_TBB ?= -I $(TBB)
```

Comment:
Could this be simplified if at the top, it was writtenas:

```
ifdef SYS_LIBS
    CXXFLAGS_TBB ?= -I $(TBB)
    LDFLAGS_TBB ?= -ltbb
    LDFLAGS_SUNDIALS ?= -lsundials_nvecserial -lsundials_cvodes -lsundials_idas -lsundials_kinsol
endif
```

just in one location? 

One that's set, the other parts shouldn't overwrite the instruction because it's using `?=`. (To be more explicit, you could just use `=` to set it.)